integration-tests/steps/stacks.py

```diff
@@ -145,6 +145,15 @@ def dump_stack_config(config_path: Path, config_dict: dict):
         yaml.safe_dump(config_dict, f)
 
 
+@given('a stack setting in log group "{log_group_name}" has drifted')
+def step_impl(context, log_group_name):
+    client = boto3.client('logs')
+    client.put_retention_policy(
+        logGroupName=log_group_name,
+        retentionInDays=7
+    )
```

Comment:
This only works if you have the exact log group name. However, that's a little inflexible for these integration tests. 
Consider this flow instead:
* The step is `'the log group in stack "{stack_name}" has drifted'`
* Implementation would be:
    1. use the full stack name
    2. describe the stack resources
    3. Get the first log group in the stack resources

I did something similar to this needing an SNS topic. See how I did it here: https://github.com/Sceptre/sceptre/pull/1153/files#diff-9d983b8766fce43f867c9a4c615f742a03f0c93be62ba7090508c18ff7ff1e75R53

Reply:
@jfalkenstein IMO that's a whole lot of extra code, that also runs much slower due to making more API calls, to have to write just to be able to get the log group name by stack name. If this behave code was not already problematic, it might seem worth it, but most of these 'steps' already have hard-coded assumptions around them that really prevent much by way of code re-use.

Reply:
Yeah, you're probably right. Especially if you move this to the "drift.py" steps module like you said you would and make name your stack group after the feature you're testing like I think you have also said you would, I'm less concerned about drift-based steps in a drift-named module for a drift-named feature testing specifics of a drift-named stack-group.

Reply:
As discussed in the end I changed this to fetch a stack output, thus avoiding a need for repeated API calls and making the code cleaner.